src/actions/BitPayActions.js

```diff
@@ -90,23 +90,26 @@ export async function launchBitPay(
 
   // Select payment wallet, if this wasn't called directly from a wallet's
   // send scene
-  let selectedWallet, currencyCode
+  let selectedWallet, selectedCurrencyCode
   if (params.wallet) {
     // Ensure the core wallet is accepted by this invoice as a payment option
-    currencyCode = params.wallet.currencyInfo.currencyCode
-    if (!paymentCurrencies.includes(currencyCode)) {
+    selectedCurrencyCode = params.wallet.currencyInfo.currencyCode
+    if (!paymentCurrencies.includes(selectedCurrencyCode)) {
       throw new BitPayError(BitPayErrorCode.InvalidPaymentOption, { text: paymentCurrencies.join(', ') })
     }
     selectedWallet = params.wallet
   } else {
     // Check if user owns any wallets that are accepted by the invoice
-    if (paymentCurrencies.length === 0) {
+    const { currencyWallets = {} } = params
+    const matchingWallets = Object.keys(currencyWallets).filter(key => paymentCurrencies.includes(currencyWallets[key].currencyInfo.currencyCode))
```

Comment:
Add a type annotation, `const matchingWallets: string[]`, since Flow seems to be having trouble here.